### PR TITLE
Delete field

### PR DESCRIPTION
**Pull request by cameronmarsh**

### CODE REVIEW

File: apps/andi/lib/andi/input_schemas/data_dictionary_fields.ex [hunk 1]
```diff
@@ -10,30 +10,30 @@ defmodule Andi.InputSchemas.DataDictionaryFields do
 
   @top_level_bread_crumb "Top Level"
 
-  def add_field_to_parent(original_field, parent_bread_crumb) do
-    updated_field = adjust_parent_details(original_field, parent_bread_crumb)
-    changeset = DataDictionary.changeset(%DataDictionary{}, updated_field)
+  def add_field_to_parent(new_field, parent_bread_crumb) do
+    new_field_updated = adjust_parent_details(new_field, parent_bread_crumb)
```
Comment (bbrewer-papa-x) [inline apps/andi/lib/andi/input_schemas/data_dictionary_fields.ex:14]: I like it

File: apps/andi/lib/andi/input_schemas/data_dictionary_fields.ex [hunk 2]
```diff
@@ -10,30 +10,30 @@ defmodule Andi.InputSchemas.DataDictionaryFields do
...
-        |> Map.put(:bread_crumb, field.name)
+    if existing_field != nil do
+      case Repo.delete(existing_field) do
+        {:error, _changeset} ->
+          {:error, DataDictionary.changeset(%DataDictionary{}, existing_field)}
 
-      _ ->
-        field
-        |> Map.put(:bread_crumb, parent_bread_crumb <> " > " <> field.name)
+        good ->
```
Comment (bbrewer-papa-x) [inline apps/andi/lib/andi/input_schemas/data_dictionary_fields.ex:34]: good

File: apps/andi/lib/andi_web/live/edit_live_view.ex [hunk 3]
```diff
@@ -359,6 +372,35 @@ defmodule AndiWeb.EditLiveView do
      )}
   end
 
+  def handle_info({:remove_data_dictionary_field_succeeded, deleted_field_parent_id, deleted_field_index}, socket) do
+    new_selected_field =
+      socket.assigns.changeset
+      |> get_new_selected_field(deleted_field_parent_id, deleted_field_index)
+
+    new_selected_field_id =
+      case new_selected_field do
```
Comment (bbrewer-papa-x) [inline apps/andi/lib/andi_web/live/edit_live_view.ex:381]: do we know if changing how this works affects that bug?

Comment (cameronmarsh) [reply apps/andi/lib/andi_web/live/edit_live_view.ex:381]: It doesn't :(

File: apps/andi/assets/css/data_dictionary_form.scss [hunk 4]
```diff
@@ -168,7 +192,7 @@ $tick-height: 1rem + $field-padding - 0.44;
     right: 0;
 }
 
-.data-dictionary-add-field-editor-form-container--visible {
+.modal-form-container {
```
Comment (bbrewer-papa-x) [inline apps/andi/assets/css/data_dictionary_form.scss:195]:
```suggestion
.modal-form-container {🍕🍕🍕
```

File: apps/andi/lib/andi_web/live/edit_live_view.ex [hunk 5]
```diff
@@ -359,6 +372,26 @@ defmodule AndiWeb.EditLiveView do
      )}
   end
 
+  def handle_info({:remove_data_dictionary_field_succeeded, deleted_field_parent_id, deleted_field_index}, socket) do
+    new_selected_field =
```
Comment (bbrewer-papa-x) [inline apps/andi/lib/andi_web/live/edit_live_view.ex:376]: You figured it out?

Comment (cameronmarsh) [reply apps/andi/lib/andi_web/live/edit_live_view.ex:376]: No just punting, I wrote another card to look at that. In the meantime I am trying to reproduce enough to maybe submit a bug report

File: apps/andi/assets/css/data_dictionary_form.scss (hunk 4 above)
Comment (cameronmarsh) [reply apps/andi/assets/css/data_dictionary_form.scss:195]: hmm

**Pull request merged**